### ISSUE TITLE
Editing PNG images (i.e. screenshots) produces images with jagged edges and washed out colors

### ISSUE THREAD

**Issue by kaoneko** [SimpleMobileTools/Simple-Gallery#2349]
Play Store release 6.22.0, also occurs with 6.22.3.
Running on a Google Pixel 4a with stock Android 11 ROM.

Original screenshot:
![Screenshot_20220210-112032](https://user-images.githubusercontent.com/26896621/153395382-2446a996-eaf0-4a22-9a19-c9a9c5c6292a.png)

Edited screenshot:
![Screenshot_20220210-112032_edited](https://user-images.githubusercontent.com/26896621/153395391-c8dab23a-1a34-44a4-8273-bdfc73df9ee0.png)

It doesn't seem to matter what kind of editing you do, e.g. adding a sticker, draw a line, add some text, apply a filter.

These closed issues also describe this behaviour:
**Large png images get distorted in editor** #1748 
**Editing image and exporting it produces low quality image** #1762

(They don't mention the colors, but you can see it in the example images.)

Both were marked as duplicates of:
**Poor image quality for cropped image** #1744 

Which was closed two years ago with the remark _should be fixed in 6.13.0 released soon_.

Is this a regression?

**Comment by MK73DS** [comment]
I'm experiencing the same issue, using Play Store v6.22.0 Pro. First image is a screenshot cropped on my PC using GIMP, second image is a screenshot cropped via the app. The quality is hugely inferior, and the colors are also off.
![Screenshot_20220510-133333](https://user-images.githubusercontent.com/10391562/167619837-945e5cf5-b3e9-40e0-8e03-350dbaa661e2.png)
![Screenshot_20220510-133333_1](https://user-images.githubusercontent.com/10391562/167619846-13e0a1b1-50f4-4c37-9624-a767f17db851.png)

This seems to happen only when editing png files, I didn't encounter the issue with jpg files. Also, sometimes the jagged edges are less noticeable depending on the cropping size and aspect ratio, but the color is always off. See the following picture, the edges are more or less right but still not as good as the original image. Note that I'm only cropping the image, no scaling is involved, therefore there shouldn't be artifacts.
![Screenshot_20220510-133333_2](https://user-images.githubusercontent.com/10391562/167620660-b0101cfb-35d5-4a98-885b-71b45e67941d.png)

It is quite annoying since it makes small text hard to read, therefore I need to use another photo editor app just to crop my screenshots before sending them.

**Comment by lipici** [comment]
Same issue at me with proprietary apk, with foss version its ok
Also, dosent matter if is png or not, its the same for every picture extension, washed out. Last version at moment of posting.

**Comment by lipici** [comment]
![Screenshot_20220526-170134_1](https://user-images.githubusercontent.com/46893370/170814609-35199c53-5894-416a-b85b-8820a1a22c44.png)
![Screenshot_20220526-170134](https://user-images.githubusercontent.com/46893370/170814611-0987db4a-9047-45fe-b6c0-f2d9cd539cfa.png)

Can be resolved this bug ?

**Comment by tp0** [comment]
Jagged lines issue when exporting png does not occur in the FOSS version.

Presumably it is a problem in the [PhotoEditorSDK](https://github.com/imgly/pesdk-android-demo) which would also explain why the problem has persisted despite multiple reports here. That Github page offers no way report issues so I guess that's handled via their customer support for license owners.

**Comment by adarsh1001** [comment]
Issue reproducible on the latest 6.25.5 Pro as well, jagged lines after cropping a PNG.

**Comment by kaoneko** [comment]
I just checked and this issue still exists in the current version of Simple Gallery Pro. I just don't understand how this doesn't get prioritized, it seems quite severe to me... @tibbi? I've switched to Google Photos for basic editing since opening this issue, which luckily can be invoked from within Simple Gallery, but I wish it wasn't necessary. I can surely see people ditching Simple Gallery altogether because of this issue.

The fact that it only seems to affect screenshots (PNG files) maybe makes it less severe, but I for one don't trust the editor with my photos anymore either now that I know what it does to screenshots.

**Comment by Asiaboy89** [comment]
Can confirm this behaviour, but this issue was not always present, only after updating the app to a newer version 1,5 years ago or so.

I'm working often with screenshots (PNGs) and small texts within the screenshots and after adding some comments/text by myself via the editor, the screenshot is noticably worse than before. When I do this 2 or 3 times with the same screenshot (means: adding text, saving photo, adding text, saving, etc.), the content of the screenshot is barely readable.

Please fix that, because it was not always like this.
